openquake/hazardlib/gsim/raghukanth_iyengar_2007.py

```diff
@@ -0,0 +1,573 @@
+    #: typographical errors in the published coefficients. In each case the
+    #: value sugstituted is interpolated from neighbouring values.
+    #:
+    #: RaghukanthIyengar2007 COEFFS_BEDROCK (Table 3) at 1.200 s:
+    #:
+    #: * change c1 from 0.2904 to 0.1904
+    #:
+    #: RaghukanthIyengar2007 COEFFS_NEHRP_C (Table 5) at 0.750 s:
+    #:
+    #: * change a1 from 0.36 to -0.30
+    #:
+    #: RaghukanthIyengar2007Southern COEFFS_BEDROCK (Table 2(b)) at 2.000 s:
+    #:
+    #: * change c4 from 0.0001 to 0.0010
+    #:
+    #: Note that these would be in addition to the following more obvious
+    #: correction which was implemented.
+    #:
+    #: RaghukanthIyengar2007Southern COEFFS_BEDROCK (Table 2(b)) at 0.150 s:
+    #:
+    #: * change c1 from .1941 to 2.1941
+    #:
+    #: Note that since test data was dervied from Figures 3 and 5, PGA is
+    #: not covered.
+    non_verified = True
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        # pylint: disable=too-many-arguments
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for specification of input and result values.
+
+        Implements the following equations:
+
+        Equation (8) on p. 203 for the bedrock ground motion:
+
+        ``ln(y_br) = c1 + c2*(M - 6) + c3*(M - 6)**2 - lnR - c4*R + ln(ε_br)``
+
+        Equation (9) on p. 207 gives the site amplification factor:
+
+        ``ln(F_s) = a1*y_br + a2 + ln(δ_site)``
+
+        Equation (10) on p. 207 for the ground motion at a given site:
+
+        ``y_site = y_br*F_s``
+
+        Equation (11) on p. 207 for total standard error at a given site:
+
+        ``σ{ln(ε_site)} = sqrt(σ{ln(ε_br)}**2 + σ{ln(δ_site)}**2)``
+
+        """
+
+        assert imt.__class__ in self.DEFINED_FOR_INTENSITY_MEASURE_TYPES
+
+        # obtain coefficients for required intensity measure type
+        coeffs = self.COEFFS_BEDROCK[imt].copy()
+
+        # obtain site-class specific coefficients
+        a_1, a_2, sigma_site = self._get_site_coeffs(sites, imt)
+        coeffs.update({'a1': a_1, 'a2': a_2, 'sigma_site': sigma_site})
+
+        # compute bedrock motion, equation (8)
+        ln_mean = (self._compute_magnitude_terms(rup, coeffs) +
+                   self._compute_distance_terms(dists, coeffs))
+
+        # adjust for site class, equation (10)
+        ln_mean += self._compute_site_amplification(ln_mean, coeffs)
+        # No need to convert to g since "In [equation (8)], y_br = (SA/g)"
+
+        ln_stddevs = self._get_stddevs(coeffs, stddev_types)
+
+        return ln_mean, [ln_stddevs]
+
+    def _compute_magnitude_terms(self, rup, coeffs):
+        """
+        First three terms of equation (8) on p. 203:
+
+        ``c1 + c2*(M - 6) + c3*(M - 6)**2``
+        """
+
+        adj_mag = rup.mag - self.CONSTS['ref_mag']
+        return coeffs['c1'] + coeffs['c2']*adj_mag + coeffs['c3']*adj_mag**2
+
+    @classmethod
```

Comment:
It is actually true that it is better to use a classmethod when the method does not depend on self,  but it is a minor thing.